src/commands/publish/upload_form/mod.rs

```diff
@@ -0,0 +1,140 @@
+mod file;
+mod project_assets;
+mod script;
+mod wasm_module;
+
+use log::info;
+
+use reqwest::multipart::{Form, Part};
+use std::fs;
+use std::path::Path;
+
+use crate::commands::build::wranglerjs::Bundle;
+use crate::settings::binding;
+use crate::settings::metadata::Metadata;
+use crate::settings::project::{Project, ProjectType};
+
+use project_assets::ProjectAssets;
+use script::Script;
+use wasm_module::WasmModule;
+
+use super::{krate, Package};
+
+pub fn build_script_upload_form(project: &Project) -> Result<Form, failure::Error> {
+    let project_type = &project.project_type;
+    match project_type {
+        ProjectType::Rust => {
+            info!("Rust project detected. Publishing...");
+            let name = krate::Krate::new("./")?.name.replace("-", "_");
+            // TODO: move into build?
+            build_generated_dir()?;
+            concat_js(&name)?;
```

Comment:
these two steps feel like they belong in the build step, however for now i'm leaving these here with a todo, can create an issue unless they should stay here.